### PR TITLE
The Witness: Panel Hunt Plando

### PR DESCRIPTION
**Pull request by NewSoupVi**
Add an option to allow the player to specify which panels they want to be hunt panels in their panel hunt game.

Also fixes some lint/typing stuff in the panel hunt tests

Tested:
This has been played as part of a beta test. Otherwise, test gens hitting every code path

### CODE REVIEW

File: worlds/witness/entity_hunt.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                self.PRE_PICKED_HUNT_ENTITIES.add(entity_hex)
+
+    def _get_eligible_panels(self) -> Tuple[List[str], Dict[str, Set[str]]]:
+        """
+        There are some entities that are not allowed for panel hunt for various technical of gameplay reasons.
+        Make a list of all the ones that *are* eligible, plus a lookup of eligible panels per area.
+        """
+
+        all_eligible_panels = [
+            panel for panel in ALL_HUNTABLE_PANELS if self._entity_is_eligible(panel) if panel not in self.HUNT_ENTITIES
```
Comment (Exempt-Medic) [inline worlds/witness/entity_hunt.py:128]: `self.HUNT_ENTITIES` is empty here

File: worlds/witness/entity_hunt.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        2. Pick random hunt entities to fill out the rest
+        3. Replace unfair entities with fair entities
+
+        Each of these is its own function.
+        """
+
+        self.HUNT_ENTITIES = self.PRE_PICKED_HUNT_ENTITIES.copy()
+
+        self._pick_all_hunt_entities(total_amount)
+        self._replace_unfair_hunt_entities_with_good_hunt_entities()
```
Comment (Exempt-Medic) [inline worlds/witness/entity_hunt.py:67]: This function should account for plando'ed Panel Hunt panels.

Comment (NewSoupVi) [reply worlds/witness/entity_hunt.py:67]: It does.
Plandoed panels are added to `self.PRE_PICKED_HUNT_ENTITIES`
In this snapshot you can see: `self.HUNT_ENTITIES = self.PRE_PICKED_HUNT_ENTITIES.copy()`
And `_replace_unfair_hunt_entities_with_good_hunt_entities()` operates on `self.HUNT_ENTITES`.

Comment (NewSoupVi) [reply worlds/witness/entity_hunt.py:67]: Oh, I just realized what you meant.
You want it to **not** replace plandoed stuff

**Review by Exempt-Medic — APPROVED** [review]
The changes LGTM. I tested various code paths code get hit and functioned properly and have been using this in a lot of my own games lately where it has been working without error. (I might review the full list of plando-able locations later but that doesn't impact this PR since hunts can already land on those panels anyway)

**Pull request merged**